### PR TITLE
Use Regexp.quote for version numbers

### PR DESCRIPTION
**Pull request by aried3r**
### Description

This was already used in [other places](https://github.com/sous-chefs/ruby_rbenv/blob/560b851a9e19a4e72000c995b00e5443f3fb87c7/test/integration/gem/controls/gem_install.rb#L10), this uses it everywhere else as well. But other places just use `include` which might be easier to read and still be a good choice for tests.

The `match` assertion would be more useful if it was extremely strict, at which point we could just use `its ('stdout') { should equal 'foo' }` to have an exact match.

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

---
This change is [Reviewable](https://reviewable.io/reviews/sous-chefs/ruby_rbenv/227)

### CODE REVIEW

**Comment by aried3r** [comment]
I can't reproduce the centos-7 failures locally running. Could you maybe restart the job?

**Comment by damacus** [comment]
Yeah ignore some of the failures for this right now, it's likely a Travis error.

I actually really dislike `Regexp.quote` syntax, it doesn't read very nicely. Happy to go with basically any other alternative like `%r{}` or something similar.

**Comment by aried3r** [comment]
> I actually really dislike Regexp.quote syntax

Yeah, it doesn't. I probably don't know all the different ways to do this, but I'd either just use `should include '2.5.1'`, because in this case it does what we want, unless we want to match the exact output.

```Ruby
# .include?
» 'Ruby version 2.5.1'.include?('2.5.1')
=> true
» 'Ruby version 2f5b1'.include?('2.5.1')
=> false

# .match
» 'Ruby version 2.5.1'.match?(/2.5.1/)
=> true
» 'Ruby version 2f5b1'.match?(/2.5.1/)
=> true
» 'Ruby version 2f5b1'.match?(/"#{Regexp.quote('2.5.1')}/)
=> false
```

File: test/integration/user_install/controls/user_install.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@
   describe bash('sudo -H -u vagrant bash -c "source /etc/profile.d/rbenv.sh && rbenv global"') do
     its('exit_status') { should eq 0 }
     its('stdout') { should include(global_ruby) }
-    its('stdout') { should_not match(/system/) }
+    its('stdout') { should_not match(/#{Regexp.quote(system)}/) }
```
Comment (damacus) [inline test/integration/user_install/controls/user_install.rb:12]: This should be `#{Regexp.quote("system")}`, making it a  string not an object (that doesn't exist).

**Review by damacus — CHANGES_REQUESTED** [review]

File: test/integration/user_install/controls/user_install.rb (hunk 1 above)
Comment (damacus) [reply test/integration/user_install/controls/user_install.rb:12]: Failure here 
https://travis-ci.org/sous-chefs/ruby_rbenv/jobs/422229016

Comment (aried3r) [reply test/integration/user_install/controls/user_install.rb:12]: Good catch. I think I'll close this PR and open another where I replace `match` with `include`, what do you think?

Comment (damacus) [reply test/integration/user_install/controls/user_install.rb:12]: I'm happy with that. if it works! :)

**Comment by aried3r** [comment]
Closing, sending a new one using `include` instead of `match` soon.

**Pull request closed without merging**